source/hardware/controllers/native_instruments_traktor_kontrol_s3.rst

```diff
@@ -0,0 +1,173 @@
+=================================  ==========================================================
+
+Looping
+~~~~~~~
+
+======================================   ================================================
+Control                                  Description
+======================================   ================================================
+Right Encoder Turn                       Double/halve loop size.
+Right Encoder Press                      Activate loop of set size from current position.
+:hwlabel:`SHIFT` + Right Encoder Press   Toggles the existing loop on and off.
+Left Encoder Turn                        Beatjump forward/backward.
+:hwlabel:`SHIFT` + Left Encoder Turn     Adjust beatjump size.
+Left Encoder Press                       Activates beatloop roll.
+:hwlabel:`SHIFT` + Left Encoder Press    Activates a loop and then stops.
+======================================   ================================================
+
+Rate / Keylock
+~~~~~~~~~~~~~~
+
+There are two ways the rate sliders can be mapped: Absolute, and Relative.  Absolute mode is the default. In this mode, the position of the pitch slider matches the on-screen pitch slider position.  If the sliders are misaligned, Mixxx engages "soft takeover mode" -- Mixxx won't update the value of the slider until the controller matches the GUI. In Relative mode, moving the slider always adjusts the value of the pitch slider, even if they don't match.
+
+  - Absolute Mode:
+
+     - :hwlabel:`KEYLOCK`: Press to toggle keylock mode.
+     - :hwlabel:`SYNC`: Press to beatsync, or press and hold to activate Sync Lock Mode.
+     - Pitch slider: Adjusts playback speed.
+     - :hwlabel:`SHIFT` + Pitch slider: Adjusts musical key
+  - Relative Mode:
+
+     - :hwlabel:`KEYLOCK`: Press to toggle keylock mode.
+     - :hwlabel:`SYNC`: Press to beatsync, or press and hold to activate Sync Lock Mode.
+     - Pitch slider: Adjusts playback speed.
+     - Keylock + Pitch Slider: adjusts musical key
+     - :hwlabel:`SHIFT` + Pitch Slider: Allows the user to move the slider without any effect.
+
+Effects
+~~~~~~~
+
+The FX setup is unusual on this controller.  Each deck has a single toggle button for effects, and on the right-hand side of the mixer there are five buttons that determine which effects are applied to every channel that has effects on.  This means it is not possible to use the controller to select one effect for one deck, and another effect for another.  You can still make these choices in the Mixxx UI, however.
+
+When the :hwlabel:`FILTER ENABLE` button is off, the knob still controls the default Quick Effect, even if that button is not lit in the FX section. When the :hwlabel:`FILTER ENABLE` button is on, the Quick Effect is only enabled if the Filter :hwlabel:`FX SELECT` button is on.  This means the Quick Effect is available on channels that don't have any other effects active.
```

Comment:
I am confused by the discrepancy between the code and the documentation here. The code seems much more complex and I see it switches to controlling the regular effects units but that's not mentioned at all here.

Reply:
this has been updated